tests/lib/eslint/flat-eslint.js

```diff
@@ -4437,7 +4468,7 @@ describe("FlatESLint", () => {
             beforeEach(prepare);
             afterEach(cleanup);
 
-            it("'lintFiles()' with a directory path should NOT contain 'foo/test.txt' and 'foo/nested/test.txt'.", async () => {
+            it("'lintFiles()' with a directory path should contain 'foo/test.txt' and 'foo/nested/test.txt'.", async () => {
```

Comment:
For some reason `foo/**/*` was NOT matching `.txt` files prior to the changes in this PR. That seems like a mistake based on how flat config is intended to work, so I updated this test.

Reply:
On second thought, I believe this change is incorrect. I was previously filtering out `/*` and I think I should still do that. I'll get to work on that tomorrow.

Reply:
Maybe we could simplify this by updating `ConfigArray#getConfig` to consider the file ignored if it doesn't match at least one pattern that doesn't end with `*`.  That would unify the behavior of directories and globs, so `eslint foo`, where `foo` is a directory, would be the same as `eslint "foo/**"`. A difference from the current design is that all targeted file extensions would need to be specified in the config file. For example, `eslint "src/**/*.ts"` would not match anything if the only config entry is `files: ["src/**"]`. That entry would have to be `files: ["src/**/*.ts"]` (or there could be another one that adds all ts files as lint targets, e.g. `files: ["**/*.ts"]`).

Reply:
Yup, that's exactly what I've implemented. Basically, the behavior of `/**` and`/*` now act as if the `files` key is missing: they only apply if something else matches.